src/Views/CategoryView/categoryView.cpp

```diff
@@ -67,37 +80,68 @@ CategoryView::CategoryView() :
 			case UTILITIES:
 				hbs = db->Filter(IsCategory("4"));
 				break;
+			case SEARCH:
+				break;
 			}
 			categoryTabs.push_back(CategoryTab(ListView(hbs)));
 		} catch (const std::exception& ex) {
 			categoryTabs.push_back(CategoryTab(ListView(std::vector<Homebrew>())));
-            log_printf(DBG_ERROR, "Couldn't create listViews: %s", ex.what());
+			log_printf(DBG_ERROR, "Couldn't create listViews: %s", ex.what());
 		}
 
 	}
 
+	log_printf(DBG_DEBUG, "Done tabs creation");
+
 
 	int remainingWidth = SCREEN_WIDTH;
-	double categoryWidth = SCREEN_WIDTH / _countof(categoryList);
-
-	categoryTabs[0].minX = 0;
-	categoryTabs[0].maxX = (int)categoryWidth;
-    log_printf(DBG_DEBUG, "0->maxX=%d", categoryTabs[0].maxX);
-	remainingWidth -= (int)categoryWidth;
-    log_printf(DBG_DEBUG, "remainingWidth=%d", remainingWidth);
-
-	for (unsigned int i=1; i < _countof(categoryList); i++) {
-		categoryTabs[i].minX = categoryTabs[i-1].maxX;
-		categoryTabs[i].maxX = categoryTabs[i].minX + (int)remainingWidth / (_countof(categoryList) - i);
-		remainingWidth -= (int)categoryTabs[i].maxX + 1 - categoryTabs[i].minX;
-        log_printf(DBG_DEBUG, "%d->minX=%d", i, categoryTabs[i].minX);
-        log_printf(DBG_DEBUG, "%d->maxX=%d", i, categoryTabs[i].maxX);
-        log_printf(DBG_DEBUG, "remainingWidth=%d", remainingWidth);
+	unsigned int countAutoWidth = 0;
+	for (unsigned int i=0; i < _countof(categoryList); i++) {
+		if (categoryList_widths[i] >= 0) {
+			remainingWidth -= categoryList_widths[i];
+		} else {
+			countAutoWidth += 1;
+		}
+	}
+
+	log_printf(DBG_DEBUG, "countAutoWidth %d", countAutoWidth);
+
+	unsigned int categoryAutoWidth = 0;
+	if (remainingWidth > 0 && countAutoWidth > 0) {
+		categoryAutoWidth = remainingWidth / countAutoWidth;
+		remainingWidth -= categoryAutoWidth * countAutoWidth;
+	}
+
+	for (unsigned int i=0; i < _countof(categoryList); i++) {
+		if (i == 0) {
+			categoryTabs[i].minX = 0;
```

Comment:
I suggest moving this base case back out of the loop.

Reply:
I'm iterating over 0 anyway or else I'd have to duplicate lines 121-132, which would be bad style, I guess.